enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,454 @@
+feature gate.
+This feature gate is also the main feature gate used to toggle the SCC changes above. This is done because the same version skew issue exists in Openshift,
+and it would be better to keep UserNamespacesSupport on by default (not skewing from upstrem).
+
+ProcMountType is not directly related to user namespaces, but it does rely on them, and is useful for podman in Openshift pod use cases. Specifically,
+it's needed to allow podman to configure networking for the sub containers (as mounting a sub '/proc' requires the whole '/proc' be read-write, and '/proc' needs
+to be writable to configure sysctls)
+
+The goal is to graduate this feature in 4.18, meaning `UserNamespacesPodSecurityStandards` and `ProcMountType` will be enabled by default, and will not mark
+a cluster as TechPreviewNoUpgrade.
+
+#### MinimalKubeletVersion in the apiserver
+
+An additional piece is needed for ensuring every node that is in a cluster with `UserNamespacesSupport` enabled actually runs those pods with user namespaces.
+Unfortunately, we cannot rely on homogenous feature gates protecting us in this case. For instance, an admin may pause a worker pool while an upgrade happens, but those workers
+can still be scheduled to. The MCO should not be enabling feature gates alpha feature gates for older kubelets, even if the `featuregate` Openshift object enables them [1].
+
+Since there can be a version skew between the kubelet and the kube-apiserver, and the feature gates are not assumed to be homogeneous, then it cannot be assumed that every
+kubelet in the cluster will be new enough to support UserNamespaces.
+
+While this would not be an issue if the featuregates are enabled on kubelet nodes after 4.15, we cannot assume the feature gate will be enabled.
+
+Thus, we need a method for the kube-apiserver to know for certain a node joining the cluster has a Kubelet that will also have the same featuregates set.
+A way to do this that also could set a precedent for Openshift to have the apiserver be more concious about kubelet version would be an openshift-apiserver
+extension that refuses to let a kubelet join a cluster or get leases if it is not new enough.
+
+For such a feature, there could be an extension to the NodeConfig object called `minimumKubeletVersion`. This field would be set on the cluster level. Practically, the
+openshift-apiserver could then query the `Node.Status.NodeSystemInfo.KubeletVersion` and verify it is above the `minimumKubeletVersion`.
+
+It is also possible this feature should be paired with a corresponding kubelet field `minimumKubeletVersion`, where it exits is it is too old. This will prevent the kubelet from
+running before it seeks to get credentials from the kube-apiserver, but also adds additional code overhead and backporting, plus given this feature would be added in z-streams, it's not
+possible to rely on.
```

Comment:
Does the node/MCO team want more detail about how this mechanism will work?  Having this working is a requirement to move UserNamespaces out of TechPreview prior to the featuregate being enabled on all kubelets (not kube-apiservers), which we also have no way to do.

Reply:
@mrunalp ?